akka-stream/src/main/scala/akka/stream/javadsl/SubSource.scala

```diff
@@ -294,6 +294,27 @@ class SubSource[+Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Source
   def collect[T](pf: PartialFunction[Out, T]): SubSource[T, Mat] =
     new SubSource(delegate.collect(pf))
 
+  /**
+   * Transform this stream by testing the type of each of the elements
+   * on which the element is an instance of the provided type as they pass through this processing step.
+   * Non-matching elements are filtered out.
+   *
+   * Adheres to the [[ActorAttributes.SupervisionStrategy]] attribute.
+   *
+   * '''Emits when''' the element is an instance of the provided type
+   *
+   * '''Backpressures when''' the element is an instance of the provided type and downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def collectType[T](clazz: Class[T]): javadsl.SubSource[T, Mat] = {
+    new SubSource(delegate.collect({
+      case elem if clazz.isInstance(elem) ⇒ clazz.cast(elem)
+    }))
+  }
```

Comment:
Why not call `collectType` on the `delegate`?

Reply:
thanks done